dashboard/src/shared/utils.ts

```diff
@@ -57,3 +61,26 @@ export function flattenResources(
   });
   return result;
 }
+
+export function getPluginIcon(plugin: Plugin | string) {
```

Comment:
Initial change towards having custom UIs based upon the plugin. I just have split removed each imported icon in favor of a centralized way to decide which icon we should render.

Reply:
Excellent.